### PR TITLE
Check for icims field on namely

### PR DESCRIPTION
**Pull request by tabfugnic**
Check for field on namely employee that allows us to link back to the user from icims

### CODE REVIEW

File: spec/models/icims/connection_spec.rb [hunk 1]
```diff
@@ -52,4 +52,31 @@
       expect(icims_connection.password).to be_nil
     end
   end
+
+  describe "#missing_namely_field?" do
+    it "is doesn't check if not connected" do
```
Comment (hrs) [inline spec/models/icims/connection_spec.rb]: Grammar!

File: spec/features/user_deletes_icims_connection_spec.rb [hunk 2]
```diff
@@ -7,6 +7,7 @@
       :icims_connection,
       :connected,
       user: user,
+      found_namely_field: true
```
Comment (hrs) [inline spec/features/user_deletes_icims_connection_spec.rb]: Trailing comma!

File: spec/models/icims/connection_spec.rb [hunk 3]
```diff
@@ -52,4 +52,31 @@
...
+      icims_connection.missing_namely_field?
+
+      expect(icims_connection).to_not have_received(:check_namely_field)
+    end
+
+    it "checks and caches required namely field status" do
+      icims_connection = create(
+        :icims_connection,
+        :connected,
+        found_namely_field: false
```
Comment (hrs) [inline spec/models/icims/connection_spec.rb]: Trailing comma!

Comment (hrs) [inline spec/models/icims/connection_spec.rb]: Also, should we be assigning `found_namely_field` to something in the factory?

File: spec/models/icims/connection_spec.rb [hunk 4]
```diff
@@ -52,4 +52,31 @@
...
   end
+
+  describe "#missing_namely_field?" do
+    it "is doesn't check if not connected" do
+      icims_connection = described_class.new
+      allow(icims_connection).to receive(:check_namely_field)
+
+      icims_connection.missing_namely_field?
+
+      expect(icims_connection).to_not have_received(:check_namely_field)
```
Comment (hrs) [inline spec/models/icims/connection_spec.rb]: `not_to`?

File: app/models/icims/connection.rb [hunk 5]
```diff
@@ -23,10 +23,35 @@ def disconnect
       )
     end
 
+    def missing_namely_field?
+      if connected?
+        check_namely_field
+        !found_namely_field?
```
Comment (elle) [inline app/models/icims/connection.rb]: Would it read better if we renamed `found_namely_field?` to `namely_field_not_found?`?
It is used twice with the `!` oeprator

Comment (elle) [inline app/models/icims/connection.rb]: A bit confusing that `found_namely_field?` appears here and inside the `check_namely_field` method

File: app/models/icims/connection.rb [hunk 6]
```diff
@@ -23,10 +23,39 @@ def disconnect
...
+      end
+      namely_field_not_found?
+    end
+
+    def namely_field_not_found?
+      !found_namely_field?
+    end
+
+    def namely_account_has_required_field?
+      namely_connection.fields.all.detect do |field|
```
Comment (elle) [inline app/models/icims/connection.rb]: Is `.all` needed here?

File: app/models/icims/connection.rb [hunk 7]
```diff
@@ -23,10 +23,35 @@ def disconnect
...
     end
+
+    def check_namely_field
+      if !found_namely_field? && namely_account_has_required_field?
+        update(found_namely_field: true)
+      end
+    end
+
+    def namely_account_has_required_field?
+      namely_connection.fields.all.detect do |field|
```
Comment (elle) [inline app/models/icims/connection.rb]: Actually, would this work?
`namely_connection.fields.any? do |field|`

File: app/models/icims/connection.rb [hunk 8]
```diff
@@ -23,10 +23,35 @@ def disconnect
...
     private
 
+    delegate :namely_connection, to: :user
+
     def encoded_credentials
       Base64.encode64("#{username}:#{password}")
     end
+
+    def check_namely_field
+      if !found_namely_field? && namely_account_has_required_field?
```
Comment (elle) [inline app/models/icims/connection.rb]: Wish that was simpler, but don't know enough to suggest something better

File: app/models/icims/connection.rb (hunk 6 above)
Comment (tabfugnic) [inline app/models/icims/connection.rb]: This is a part of the namely gem that we are using, all forces it into an array

File: spec/features/user_visits_dashboard_spec.rb [hunk 9]
```diff
@@ -51,4 +48,50 @@
...
+
+  context "with a iCIMS connection, but no iCIMS field on Namely" do
+    scenario do
+      stub_namely_request("fields_without_icims")
+      user = create(:user)
+      create(
+        :icims_connection,
+        :connected,
+        user: user,
+        found_namely_field: false,
```
Comment (elle) [inline spec/features/user_visits_dashboard_spec.rb]: Do we need to define it if it is `false` by default?

Comment (tabfugnic) [inline spec/features/user_visits_dashboard_spec.rb]: despite it being the default, I'm putting it in there just so there is no ambiguity on what is being tested

**Comment by elle** [comment]
Other than those comments, it looks good to me

**Pull request merged**